nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitText.java

```diff
@@ -150,548 +145,320 @@
+        this.maxSplitSize = context.getProperty(FRAGMENT_MAX_SIZE).isSet()
+                ? context.getProperty(FRAGMENT_MAX_SIZE).asDataSize(DataUnit.B).longValue() : Long.MAX_VALUE;
+        this.lineCount = context.getProperty(LINE_SPLIT_COUNT).asInteger();
+        this.headerLineCount = context.getProperty(HEADER_LINE_COUNT).asInteger();
+        this.headerMarker = context.getProperty(HEADER_MARKER).getValue();
     }
 
+    /**
+     * Will split the incoming stream releasing all splits as FlowFile at once.
+     */
     @Override
-    public void onTrigger(final ProcessContext context, final ProcessSession session) {
-        final FlowFile flowFile = session.get();
-        if (flowFile == null) {
-            return;
-        }
-
-        final ComponentLog logger = getLogger();
-        final int headerCount = context.getProperty(HEADER_LINE_COUNT).asInteger();
-        final int maxLineCount = (context.getProperty(LINE_SPLIT_COUNT).asInteger() == 0)
-                ? Integer.MAX_VALUE : context.getProperty(LINE_SPLIT_COUNT).asInteger();
-        final long maxFragmentSize = context.getProperty(FRAGMENT_MAX_SIZE).isSet()
-                ? context.getProperty(FRAGMENT_MAX_SIZE).asDataSize(DataUnit.B).longValue() : Long.MAX_VALUE;
-        final String headerMarker = context.getProperty(HEADER_MARKER).getValue();
-        final boolean includeLineDelimiter = !context.getProperty(REMOVE_TRAILING_NEWLINES).asBoolean();
-
-        final AtomicReference<String> errorMessage = new AtomicReference<>(null);
-        final ArrayList<SplitInfo> splitInfos = new ArrayList<>();
-
-        final long startNanos = System.nanoTime();
-        final List<FlowFile> splits = new ArrayList<>();
-        session.read(flowFile, new InputStreamCallback() {
-            @Override
-            public void process(final InputStream rawIn) throws IOException {
-                try (final BufferedInputStream bufferedIn = new BufferedInputStream(rawIn);
-                        final ByteCountingInputStream in = new ByteCountingInputStream(bufferedIn)) {
-
-                    long bufferedPartialLine = 0;
-
-                    // if we have header lines, copy them into a ByteArrayOutputStream
-                    final ByteArrayOutputStream headerStream = new ByteArrayOutputStream();
-                    // Determine the number of lines of header, priority given to HEADER_LINE_COUNT property
-                    int headerInfoLineCount = 0;
-                    if (headerCount > 0) {
-                        headerInfoLineCount = headerCount;
-                    } else {
-                        if (headerMarker != null) {
-                            headerInfoLineCount = countHeaderLines(in, headerMarker);
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile != null) {
```

Comment:
We should fail-fast here and follow the existing convention of using `if (flowFile == null) { return; }` rather than indenting the entire method.